File: application/src/session/context.rs
use std::path::PathBuf;

use crate::llm::ChatMessage;

pub struct SessionContext {
    pub id: String,
    pub dir: PathBuf,
    pub history: Vec<ChatMessage>,
}

impl SessionContext {
    pub fn new(id: String, dir: PathBuf) -> Self {
        Self {
            id,
            dir,
            history: Vec::new(),
        }
    }

    pub fn add_message(&mut self, msg: ChatMessage) {
        self.history.push(msg);
    }

    pub fn memory_path(&self) -> PathBuf {
        self.dir.join("memory.md")
    }

    pub fn history_path(&self) -> PathBuf {
        self.dir.join("history.json")
    }
}

File: application/src/session/context.ts
import { ContextManager } from '../orchestrator/context-manager.js';
import { MemoryManager } from '../memory/index.js';
import type { LLMClient } from '../llm/index.js';

export interface SessionContext {
  contextManager: ContextManager;
  memory: MemoryManager;
}

export function createSessionContext(
  memoryPath: string,
  llmClient?: LLMClient,
): SessionContext {
  return {
    contextManager: new ContextManager(llmClient),
    memory: new MemoryManager(memoryPath),
  };
}
